packages/app/src/specs/banners/TrackedBanner.vue

```diff
@@ -43,15 +46,33 @@ mutation TrackedBanner_SetProjectState($value: String!) {
 }
 `
 
+gql`
+mutation TrackedBanner_recordBannerSeen($campaign: String!, $messageId: String!, $medium: String!) {
+  recordEvent(campaign: $campaign, messageId: $messageId, medium: $medium)
+}
+`
+
 const props = withDefaults(defineProps<TrackedBannerComponentProps>(), {})
 
 const emit = defineEmits<TrackedBannerComponentEmits>()
 
 const stateQuery = useQuery({ query: TrackedBanner_ProjectStateDocument })
 const setStateMutation = useMutation(TrackedBanner_SetProjectStateDocument)
+const reportSeenMutation = useMutation(TrackedBanner_RecordBannerSeenDocument)
+const bannerInstanceId = ref(nanoid())
+
+const recordBannerShown = useDebounceFn(() => {
```

Comment:
Why the need for the debounce here?  Does it take time for the `props.modelValue` to settle or something?

Reply:
The `watchEffect` was firing twice, not sure exactly which of its dependencies was updating to cause the second firing. The only solutions I could think of were to do the debounce or move this to a lifecycle hook (`onMounted`). The lifecycle hook wouldn't catch post-mount changes to props (like a delayed resolution of `hasBannerBeenShown`) so this seemed like the best solution.

The `messageId` would allow us to filter out the duplicates on the backend, but it's better to avoid sending duplicates in the first place if there's an easy way not to